firmware/src/hbridge.c

```diff
@@ -209,21 +210,26 @@ float PI(float r, float y){
     return u;
 }
 
-#define STR_MAX_ANGLE 900.0f
-#define STR_MIN_ANGLE -900.0f
+#define STR_MAX_ANGLE 135.0f
+#define STR_MIN_ANGLE -135.0f
 #define D_MIN 0.25f
 
-void hbridge_control(float angle)
+void hbridge_control(float angle) // 0 - 294.54
 {
+
+    // str_whl_position  // 0 ~ 270.26
     float dt;
-    float position_setpoint = str_whl_position - 135;
-    angle -= 135;
+    float position_setpoint = str_whl_position - 135; // 0 ~ 135.26 
+    angle -= 135;  // 0 - 159.54
+
+    if (position_setpoint > STR_MAX_ANGLE) position_setpoint = STR_MAX_ANGLE; // STR_MAX_ANGLE should be +135
+    if (position_setpoint < STR_MIN_ANGLE) position_setpoint = STR_MIN_ANGLE; // STR_MIN_ANGLE should be -135
 
-    if (position_setpoint > STR_MAX_ANGLE) position_setpoint = STR_MAX_ANGLE;
-    if (position_setpoint < STR_MIN_ANGLE) position_setpoint = STR_MIN_ANGLE;
+    if (angle > STR_MAX_ANGLE) angle = STR_MAX_ANGLE; // STR_MAX_ANGLE should be +135
+    if (angle < STR_MIN_ANGLE) angle = STR_MIN_ANGLE; // STR_MIN_ANGLE should be -135
```

Comment:
This is important to reduce the error accumulation of the PI integrator. If the values entering the PI are not in the same range/scale, it will cause a bias on the error variable, which can end up being unfixable in runtime, since we never reset the integrator.